lwm2m-client/src/main.c

```diff
@@ -0,0 +1,452 @@
+    case SERVER_OBJ_LIFETIME_RES_ID:
+        //printf("read_server_obj case 2\n");
+        SOL_LWM2M_RESOURCE_INIT(r, res, res_id, 1,
+            SOL_LWM2M_RESOURCE_DATA_TYPE_INT, (int64_t)LIFETIME);
+        break;
+    case SERVER_OBJ_BINDING_RES_ID:
+        //printf("read_server_obj case 3\n");
+        SOL_LWM2M_RESOURCE_INIT(r, res, res_id, 1,
+            SOL_LWM2M_RESOURCE_DATA_TYPE_STRING,
+            sol_str_slice_from_str("U"));
+        break;
+    default:
+        if (res_id >= 2 && res_id <= 6)
+            r = -ENOENT;
+        else
+            r = -EINVAL;
+    }
+
+    return r;
+}
+
+static int
+execute_server_obj(void *instance_data, void *user_data,
+    struct sol_lwm2m_client *client, uint16_t instance_id,
+    uint16_t res_id, const struct sol_str_slice args)
+{
+    if (res_id != SERVER_OBJ_REGISTRATION_UPDATE_RES_ID)
+        return -EINVAL;
+
+    return sol_lwm2m_send_update(client);
+}
+
+static int
+del_location_obj(void *instance_data, void *user_data,
+    struct sol_lwm2m_client *client, uint16_t instance_id)
+{
+    struct location_obj_instance_ctx *instance_ctx = instance_data;
+    bool *has_location_instance = user_data;
+
+    if (instance_ctx->timeout)
+        sol_timeout_del(instance_ctx->timeout);
+    free(instance_ctx->latitude);
+    free(instance_ctx->longitude);
+    free(instance_ctx);
+    *has_location_instance = false;
+    return 0;
+}
+
+static const struct sol_lwm2m_object location_object = {
+    SOL_SET_API_VERSION(.api_version = SOL_LWM2M_OBJECT_API_VERSION, )
+    .id = LOCATION_OBJ_ID,
+    .create = create_location_obj,
+    .read = read_location_obj,
+    .del = del_location_obj,
+    .resources_count = 6
+};
+
+static const struct sol_lwm2m_object security_object = {
+    SOL_SET_API_VERSION(.api_version = SOL_LWM2M_OBJECT_API_VERSION, )
+    .id = SECURITY_SERVER_OBJ_ID,
+    .resources_count = 12,
+    .read = read_security_server_obj
+};
+
+static const struct sol_lwm2m_object server_object = {
+    SOL_SET_API_VERSION(.api_version = SOL_LWM2M_OBJECT_API_VERSION, )
+    .id = SERVER_OBJ_ID,
+    .resources_count = 9,
+    .read = read_server_obj,
+    .execute = execute_server_obj
+};
+
+static bool
+setup_client(void)
+{
+    struct sol_lwm2m_client *client;
+    static const struct sol_lwm2m_object *objects[] =
+    { &security_object, &server_object, &location_object, NULL };
+    static bool has_location_instance = false;
+    bool ret = false;
+    int r;
+
+    srand(time(NULL));
+
+    client = sol_lwm2m_client_new("lwm2m-client", NULL, NULL, objects,
+        &has_location_instance);
+
+    if (!client) {
+        fprintf(stderr, "Could not the create the LWM2M client\n");
+        goto exit;
+    }
+
+    r = sol_lwm2m_add_object_instance(client, &server_object, NULL);
+    if (r < 0) {
+        fprintf(stderr, "Could not add a server object instance\n");
+        goto exit_del;
+    }
+
+    r = sol_lwm2m_add_object_instance(client, &security_object, NULL);
```

Comment:
You mean like the last code proposed as in:

```
+/**
+ * @brief Registers attributes into the GATT database
+ *
+ * @param attrs Array of the attributes to be registered
+ * @param len Length of the array of attributes
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_gatt_register_attributes(struct sol_gatt_attr *attrs, uint16_t len);
```
? That could work here, I guess.